application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,111 @@
+     *
+     * @param title the title of the log embed
+     *
+     * @param description the description of the log embed
+     *
+     * @param author the user to be added to the embed
+     *
+     * @param timestamp the timestamp to be added to the embed
+     *
+     * @param guild the guild to write this log to
+     *
+     * @param attachments attachments that'll be added to the message
+     */
+    public static void write(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild,
+            @NotNull Attachment... attachments) {
+        Optional<TextChannel> auditLogChannel = getAndHandleModAuditLogChannel(guild);
+        if (auditLogChannel.isEmpty()) {
+            return;
+        }
+
+        MessageAction message = auditLogChannel.orElseThrow()
+            .sendMessageEmbeds(new EmbedBuilder().setTitle(title)
+                .setDescription(description)
+                .setAuthor(author.getAsTag(), null, author.getAvatarUrl())
+                .setTimestamp(timestamp)
+                .setColor(EMBED_COLOR)
+                .build());
+
+        for (Attachment attachment : attachments) {
+            message = message.addFile(attachment.getContentRaw(), attachment.name());
+        }
+        message.queue();
+    }
+
+    /**
+     * Gets the channel used for moderation audit logs, if present. If the channel doesn't exist,
+     * this method will return an empty optional, and a warning message will be written.
+     *
+     * @param guild the guild to look for the channel in
+     */
+    public static Optional<TextChannel> getAndHandleModAuditLogChannel(@NotNull Guild guild) {
+        if (auditLogChannelNamePattern == null) {
+            auditLogChannelNamePattern =
+                    Pattern.compile(Config.getInstance().getModAuditLogChannelPattern());
+        }
+
+        Optional<TextChannel> channel = guild.getTextChannelCache()
+            .stream()
+            .filter(c -> auditLogChannelNamePattern.asMatchPredicate().test(c.getName()))
+            .findAny();
+
+        if (channel.isEmpty()) {
+            logger.warn(
+                    "Unable to log moderation events, did not find a mod audit log channel matching the configured pattern '{}' for guild '{}'",
+                    Config.getInstance().getModAuditLogChannelPattern(), guild.getName());
+        }
+        return channel;
+    }
+
+    /**
+     * Represents attachment to messages, as for example used by
+     * {@link MessageAction#addFile(File, String, AttachmentOption...)}.
+     * 
+     * @param name the name of the attachment, example: {@code "foo.md"}
+     * @param content the content of the attachment
+     */
+    public record Attachment(@NotNull String name, @NotNull String content) {
+        /**
+         * Gets the content raw, interpreted as UTF-8.
+         * 
+         * @return the raw content of the file as a {@code byte[]}.
```

Comment:
* It is obsolete to write the type here. just `return the raw content of the file` is enough
* that said, its an attachment, not a file. change the description

Reply:
done.